### PR TITLE
Api generic equipments

### PR DESCRIPTION
**Pull request by ElMehdiTailassan**

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 1]
```diff
@@ -596,140 +596,17 @@ public String getSubstationsGraphics(UUID networkUuid, UUID nodeUuid, List<Strin
...
-                substationsIds, "voltage-levels");
+            equipmentType, equipmentId);
     }
 
-    public String getVoltageLevelsAndEquipment(UUID studyUuid, UUID nodeUuid, List<String> substationsIds, boolean inUpstreamBuiltParentNode) {
+    public String getEquipmentsMapData(UUID studyUuid, UUID nodeUuid, List<String> equipmentsIds, boolean inUpstreamBuiltParentNode, String equipmentsType) {
         UUID nodeUuidToSearchIn = getNodeUuidToSearchIn(nodeUuid, inUpstreamBuiltParentNode);
         return networkMapService.getEquipmentsMapData(networkStoreService.getNetworkUuid(studyUuid), networkModificationTreeService.getVariantId(nodeUuidToSearchIn),
-                substationsIds, "voltage-levels-equipments");
+            equipmentsIds, equipmentsType);
```
Comment (Walid-Sahnoun) [inline src/main/java/org/gridsuite/study/server/service/StudyService.java:609]: would be better change name "equipmentsType" with "type" only ?

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 2]
```diff
@@ -314,268 +314,30 @@ public ResponseEntity<String> getSubstationGraphics(
...
-            @Parameter(description = "Should get in upstream built node ?") @RequestParam(value = "inUpstreamBuiltParentNode", required = false, defaultValue = "true") boolean inUpstreamBuiltParentNode) {
-
-        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(studyService.getLinesMapData(studyUuid, nodeUuid, substationsIds, inUpstreamBuiltParentNode));
-    }
-
-    @GetMapping(value = "/studies/{studyUuid}/nodes/{nodeUuid}/network-map/lines/{lineId}")
-    @Operation(summary = "Get specific line description")
-    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The line data")})
-    public ResponseEntity<String> getLineMapData(
+    @GetMapping(value = "/studies/{studyUuid}/nodes/{nodeUuid}/network-map/{equipmentType}/{equipmentId}")
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/StudyController.java:317]: Use a RequestParam for equipmentType like :
[https://github.com/gridsuite/study-server/pull/338](https://github.com/gridsuite/study-server/pull/338)

**Comment by SlimaneAmar** [comment]
I think you should rebase your PR from [https://github.com/gridsuite/study-server/pull/338](https://github.com/gridsuite/study-server/pull/338)
The work has already done for equipmentsIds

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 3]
```diff
@@ -314,268 +314,30 @@ public ResponseEntity<String> getSubstationGraphics(
...
+            @Parameter(description = "equipment type") @PathVariable("equipmentType") String equipmentType,
             @Parameter(description = "Should get in upstream built node ?") @RequestParam(value = "inUpstreamBuiltParentNode", required = false, defaultValue = "true") boolean inUpstreamBuiltParentNode) {
 
-        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(studyService.getLineMapData(studyUuid, nodeUuid, lineId, inUpstreamBuiltParentNode));
+        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(studyService.getEquipmentMapData(studyUuid, nodeUuid, equipmentId, inUpstreamBuiltParentNode, equipmentType));
     }
 
-    @GetMapping(value = "/studies/{studyUuid}/nodes/{nodeUuid}/network-map/substations")
-    @Operation(summary = "Get Network substations description")
+    @GetMapping(value = "/studies/{studyUuid}/nodes/{nodeUuid}/network-map/{equipmentType}")
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/StudyController.java:330]: /studies/{studyUuid}/nodes/{nodeUuid}/network-map/equipments?equipmentType=value

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 4]
```diff
@@ -596,140 +596,17 @@ public String getSubstationsGraphics(UUID networkUuid, UUID nodeUuid, List<Strin
...
-    public String getLoadsMapData(UUID studyUuid, UUID nodeUuid, List<String> substationsIds,
-                                  boolean inUpstreamBuiltParentNode) {
-        UUID nodeUuidToSearchIn = getNodeUuidToSearchIn(nodeUuid, inUpstreamBuiltParentNode);
-        return networkMapService.getEquipmentsMapData(networkStoreService.getNetworkUuid(studyUuid), networkModificationTreeService.getVariantId(nodeUuidToSearchIn),
-                substationsIds, "loads");
-    }
-
-    public String getLoadMapData(UUID studyUuid, UUID nodeUuid, String loadId, boolean inUpstreamBuiltParentNode) {
+    public String getEquipmentMapData(UUID studyUuid, UUID nodeUuid, String equipmentId,
+                                      boolean inUpstreamBuiltParentNode, String equipmentType) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/StudyService.java:600]: `equipmentType` and `equipmentPath` are not the same

**Comment by SlimaneAmar** [comment]
You need modify tests ?

**Comment by SlimaneAmar** [comment]
`equipmentType` and `equipmentPath` are not the same
See  [https://github.com/gridsuite/study-server/pull/338](https://github.com/gridsuite/study-server/pull/338)

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

**Pull request closed without merging**